Add: Ground sprite constants for  partially cleared tiles

rexxars opened the pull request:
This PR adds two new ground sprite constants for "partially cleared" tiles, eg the ones that will gradually appear with more grass over time after bulldozing a tile to the fully cleared state.

I wanted to reuse these tiles for a "swamp" setting that I'm building, and figured it was more of an oversight than a feature that they were not part of the constants. Correct me if I'm wrong!

planetmaker reviewed (changes requested):
If you compare the constant names to that of snow, a few lines further down, this PR is inconsistent:
there are 4 snowy stages, the fully snowed being GROUNDSPRITE_SNOW_4_4 which is duped as GROUNDSPRITE_SNOW.

Consequently with 3 clearing stages, these omitted constants should be
GROUNDSPRITE_CLEARED_1_3
GROUNDSPRITE_CLEARED_2_3
and
GROUNDSPRITE_CLEARED_3_3 as dupe to GROUNDSPRITE_CLEARED

(or should the latter be 1_3 and the other ones shifted one up? I don't know the order which is cleared 'more'. The one shown immediately after clearing IMHO should be the first, thus 1_3)

rexxars commented:
There's also `GROUNDSPRITE_DESERT_1_2` but no `GROUNDSPRITE_DESERT_2_2`, which also seems inconsistent with the snow one. Should I add one for that too, to be more consistent?

rexxars commented:
Which one of these do you prefer, @planetmaker ?

## a:
![image](https://user-images.githubusercontent.com/48200/112333966-49612700-8c78-11eb-8f05-a613cd1bc8cb.png)

## b:
![image](https://user-images.githubusercontent.com/48200/112334003-51b96200-8c78-11eb-90d5-d566f5553b82.png)

frosch123 commented:
My preferences:
| name | number | status | my preference |
| --- | --- | --- | --- |
| CLEARED | 3924 | exists | |
| GRASS_1_3 | 3943 | new | Strong preference: GRASS_x_y is better than CLEARED_x_y. |
| GRASS_2_3 | 3962 | new | Strong preference: GRASS_x_y is better than CLEARED_x_y. |
| GRASS_3_3 | 3981 | new | Weak preference: add for consistency. |
| GRASS  | 3981 | new | Weak preference: "NORMAL" is weird. |
| NORMAL | 3981 | exists | |
| SNOW_1_4 | 4493 | exists | |
| SNOW_2_4 | 4512 | exists | |
| SNOW_3_4 | 4531 | exists | |
| SNOW_4_4 | 4550 | exists | |
| SNOW | 4550 | exists | |
| DESERT_1_2 | 4512 | exists | |
| DESERT_2_2 | 4550 | new | Weak preference: add for consistency. |
| DESERT | 4550 | exists | |

rexxars commented:
Yeah that makes much more sense, @frosch123!
I've changed the constants in line with your suggestion and added the `GRASS` and `DESERT_2_2` for consistency.
